std/encoding/codegen/fields_struct.go

```diff
@@ -147,8 +147,8 @@ func (f *StructField) GenSkipProcess() (string, error) {
 }
 
 func (f *StructField) GenReadFrom() (string, error) {
-	return fmt.Sprintf(
-		"value.%[1]s, err = context.%[1]s_context.Parse(reader.Delegate(int(l)), ignoreCritical)",
-		f.name,
-	), nil
+	code := fmt.Sprintf("drdr := reader.Delegate(int(l))\n")
```

Comment:
If you create a new variable, I think it is better to surround it with `{...}` to avoid potential conflict, like: https://github.com/named-data/ndnd/blob/e5c2033375770932ea049a5bde9a9c1db7b6dcce/std/encoding/codegen/fields_natural.go#L57-L61